packages/in_app_purchase/in_app_purchase_platform_interface/lib/src/in_app_purchase_platform_addition.dart

```diff
@@ -2,18 +2,33 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'package:in_app_purchase_platform_interface/in_app_purchase_platform_interface.dart';
+
 // ignore: avoid_classes_with_only_static_members
 /// The interface that platform implementations must implement when they want to
-/// provide platform specific in_app_purchase features.
+/// provide platform-specific in_app_purchase features.
+///
+/// Platforms that wants to introduce platform-specific public APIs should create
+/// a class that either extend or implements [InAppPurchasePlatformAddition]. Then replace
+/// the [InAppPurchasePlatformAddition.instance] with an instance of that class.
+///
+/// All the APIs added by [InAppPurchasePlatformAddition] implementers will be accessed from
+/// [InAppPurchasePlatformAdditionProvider.getPlatformAddition] by the client APPs.
+/// To avoid client APPs to directly call the [InAppPurchasePlatform] APIs, an [InAppPurchasePlatformAddition] implementation must not
```

Comment:
"To avoid clients directly calling [InAppPurchasePlatform] APIs, an [InAppPurchasePlatformAddition] implementation should not be"

Reply:
done